File: solutions/javascript/pig-latin/1/pig-latin.js
export class translator {
  static word(word) {
    if (/^thr|^sch|[^aeiou]qu/.test(word))
      return word.substring(3, word.length) + word.substring(0, 3) + "ay";

    if (/^ch|^qu|^th|^rh/.test(word))
      return word.substring(2, word.length) + word.substring(0, 2) + "ay";

    if (/^[aeiou]|^yt|^xr/.test(word)) {
      return word + "ay";
    }

    return word.substring(1, word.length) + word.charAt(0) + "ay";
  }

  static translate(phrase) {
    return phrase
      .split(" ")
      .map((word) => translator.word(word))
      .join(" ");
  }
}

File: solutions/python/pig-latin/1/pig_latin.py
vowels = {'a', 'e', 'i', 'o', 'u'}

def translate_single_word(text):
    if text[0:3] in {"thr", "sch"}:
        return text[3:] + text[0:3] + "ay"
    elif text[1:3] == "qu" and text[0].isalpha() and text[0] not in vowels:
        return text[3:] + text[0:3] + "ay"
    elif text[0:2] in {"ch", "qu", "th", "rh"}:
        return text[2:] + text[0:2] + "ay"
    elif text[0:2] in {"xr", "yt"} or text[0] in vowels:
        return text + "ay"
    else:
        return text[1:] + text[0] + "ay"

def translate(text):
    return " ".join(translate_single_word(word) for word in text.split(" "))
